### PR TITLE
all: changes for the upcoming wit-bindgen-go next release

### PR DESCRIPTION
**Pull request by deadprogram**
This PR contains changes for compatibility with the upcoming next release for `wit-bindgen-go`

### CODE REVIEW

File: wit/mat.wit [hunk 1]
```diff
@@ -2,7 +2,7 @@ package wasm:cv;
 /// mat resource is a matrix of bytes, wrapper around the cv::Mat type.
 ///
 interface mat {
-    use types.{mix-max-loc-result};
+    use types.{mix-max-loc-result, rect};
 
     enum mattype {
```
Comment (ydnar) [inline wit/mat.wit:6]: This could be %type, which would resolve to Type in Go.

File: wit/mat.wit [hunk 2]
```diff
@@ -21,6 +21,9 @@ interface mat {
         /// Create a new Mat with the specified size and type.
         new-mat-with-size: static func(cols: u32, rows: u32, mattype: mattype) -> mat;
 
+        /// Clone returns a cloned full copy of the Mat.
```
Comment (ydnar) [inline wit/mat.wit:23]: I can't leave a comment above, but resource types can have constructors and an implicit resource-drop method, which could take the place of close().

**Review by ydnar — COMMENTED** [review]
Phone review. I didn't see a WIT world?

**Comment by deadprogram** [comment]
See world here: https://github.com/wasmvision/wasmcv/blob/wit-bindgen-go-next/wit/world.wit

File: wit/mat.wit (hunk 2 above)
Comment (deadprogram) [reply wit/mat.wit:23]: To match the semantics of how the OpenCV `Mat` type is handled that is how we currently handle it in the GoCV bindings with a separate `Close()` method.

File: wit/mat.wit (hunk 1 above)
Comment (deadprogram) [reply wit/mat.wit:6]: I initially had it that way, but changed it to avoid confusion.

**Pull request merged**